### PR TITLE
[CPDLP-2104] Add POST /api/v3/partnerships/ecf endpoint

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-2104

### Changes proposed in this pull request

* Added new endpoint to create partnership `/api/v3/partnerships/ecf`

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -420,6 +420,25 @@ en:
...
+              blank: "The attribute '#/cohort' must be included as part of partnership confirmations."
+              invalid: "The '#/cohort' you have entered is invalid. Enter a valid '#/cohort' attribute in the request body for partnership confirmations."
+            school_id:
+              blank: "The attribute '#/school_id' must be included as part of partnership confirmations."
+              invalid: "The '#/school_id' you have entered is invalid. Enter a valid '#/school_id' value in the request body for partnership confirmations. Contact the DfE for support if you are unable to find the '#/school_id'."
+              funding_error: "The school you have entered has not registered to deliver training using a DfE-funded training provider."
+              ineligible_error: "The school you have entered is currently ineligible for DfE funding."
+              already_confirmed: "The school you have entered is already confirmed."
+              recruited_by_other_provider: "Another provider is already confirmed to be in partnership with the school for the entered cohort. Contact the school for more information."
+            lead_provider_id:
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply config/locales/en.yml:435]: @cwrw 
I was thinking if the service class was used directly (outside of the controller), should still validate for completeness.

File: app/services/partnerships/create_partnership.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        school_id:,
+        cohort_id: cohort_rec.id,
+        delivery_partner_id:,
+        lead_provider_id:,
+      )
+    end
+
+  private
+
+    def cohort_rec
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply app/services/partnerships/create_partnership.rb:32]: Yes, definitely confusing. I didn't want to change it from the API params. I'll rename it to `cohort_record`

File: app/services/partnerships/create_partnership.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Partnerships
+  class CreatePartnership
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply app/services/partnerships/create_partnership.rb:4]: Will update

[the remark this answers is not in the document]
Comment (mooktakim) [reply app/services/partnerships/create_partnership.rb:4]: Will update

File: config/locales/en.yml [hunk 4]
```diff
@@ -420,6 +420,25 @@ en:
...
+          attributes:
+            cohort:
+              blank: "The attribute '#/cohort' must be included as part of partnership confirmations."
+              invalid: "The '#/cohort' you have entered is invalid. Enter a valid '#/cohort' attribute in the request body for partnership confirmations."
+            school_id:
+              blank: "The attribute '#/school_id' must be included as part of partnership confirmations."
+              invalid: "The '#/school_id' you have entered is invalid. Enter a valid '#/school_id' value in the request body for partnership confirmations. Contact the DfE for support if you are unable to find the '#/school_id'."
+              funding_error: "The school you have entered has not registered to deliver training using a DfE-funded training provider."
+              ineligible_error: "The school you have entered is currently ineligible for DfE funding."
+              already_confirmed: "The school you have entered is already confirmed."
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply config/locales/en.yml:433]: Will update

File: app/services/partnerships/create_partnership.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      @delivery_partner ||= DeliveryPartner.find_by(id: delivery_partner_id)
+    end
+
+    def delivery_partner_has_provider_relationship?
+      delivery_partner.provider_relationships.where(cohort: cohort_rec, lead_provider:).exists?
+    end
+
+    def validate_cohort
+      unless cohort_rec
+        errors.add(:cohort, :invalid)
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply app/services/partnerships/create_partnership.rb:54]: will update

**Review by cwrw — APPROVED** [review]
🙌

**Pull request merged**